scripts/get_orphaned_functions.py

```diff
@@ -0,0 +1,243 @@
+
+    for directory in search_directories:
+        for root, dirs, files in os.walk(directory):
+            if (
+                root.__contains__(func_location_module)
+                or root.__contains__("/man")
+                or root.__contains__("/vignettes")
+            ):
+                # print(f"{color_codes[13]} Skipping the module directory {root}{ENDC}")
+                continue
+            else:
+                # print(f"{color_codes[14]} Searching for the function in the directory {root}{ENDC}")
+                for file in files:
+                    if file.endswith(".R"):
+                        file_path = os.path.join(root, file)
+                        with open(file_path, "r") as file:
+                            file_content = file.read()
+                            if function_pattern.search(file_content):
+                                # print(f'{color_codes[12]} Function "{function_name}" is utilized EXTERNALLY in file: {file_path}{ENDC}')
+                                return True
+    return False
+
+
+def print_in_batches(list_to_print, label):
+    # print(label)
+    color_index = 0  # Start with the first color
+    for i in range(0, len(list_to_print), 5):
+        batch = list_to_print[i : i + 5]
+        # Get the current color
+        color = color_codes[color_index % len(color_codes)]
+        # Print the batch in the current color
+        print(color + ", ".join(map(str, batch)) + ENDC)
+        # Move to the next color for the next batch
+        color_index += 1
+
+
+def add_to_csv(list_1, list_2, csv_file_path):
+    with open(csv_file_path, mode="w", newline="") as file:
+        writer = csv.writer(file)
+
+        # Writing the header for list_1
+        writer.writerow(["Purely Orphaned functions", "module", "function_name"])
+
+        # Writing data from list_1
+        for module, function in list_1:
+            writer.writerow(["", module, function])
+
+        # Writing the header for list_2
+        writer.writerow(["Externally Utilized Orphaned Functions"])
+
+        # Writing data from list_2
+        for func, module in list_2:
+            writer.writerow(["", func, module])
+
+
+def main():
+    # Define the path to the modules directory
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    modules_dir = os.path.join(script_dir, "..", "modules")
+
+    print(f"{color_codes[2]}Modules List: {module_list}{ENDC}")
+    # Iterate over all modules
+    for module_name in module_list:
+
+        # Construct the path to the R directory of the module
+        module_r_code_path = os.path.join(modules_dir, module_name, "R")
+
+        # Print the module name and path
+        # print(color_codes[1] + "Processing module:", module_name, "\n", ENDC)
+        # print(color_codes[6] + "Module path:", module_r_code_path, "\n", ENDC)
+
+        # List all functions in the module
+        functions = list_functions_in_module(module_r_code_path)
```

Comment:
For me this fails here on modules/data.mining, which has code in `inst` but no `R` directory.

Reply:
Easiest fix is probably to add data.mining to ignored_modules

Reply:
Even better if we recursively recall the function to scan within our `inst` directory too, maybe by passing the directory to be scanned as a function requirement. Maybe we can term this as `scanning_directory` which may be `/R`, `/inst` or any other directory we want to include within it.
To remove redundancy for duplicate count, we can replace our `list` by a `set`.

Reply:
For now, I've moved `data.mining` to `ignored_modules`